### PR TITLE
Move Centaur tests requiring Vault to CircleCI [BT-97]

### PR DESCRIPTION
**Pull request by grsterin**

### CODE REVIEW

File: src/ci/bin/test.inc.sh
Comment (kshakir) [inline src/ci/bin/test.inc.sh:818]: ToL: I'm surprised `sudo` doesn't work on circle, or is this some other inconsistency like needing to `update-alternatives` for `pip`? Assuming circle does have sudo, something generic like `sudo -H "$(command -v pip)" install "${pip_package}" "$@"` should work on all CI systems 🤞

File: src/ci/bin/test_slurm.inc.sh
Comment (kshakir) [inline src/ci/bin/test_slurm.inc.sh:47]: Tol: Leave this list alphabetized so it doesn't end up in a random order, and use `localhost` the same as the other names above like the machine and node.

Or, adjust those names to make them consistent with your addition.

**Review by kshakir — APPROVED** [review]
A couple "ToL" comments, but generally LGTM if CI is happy

File: src/ci/bin/test.inc.sh
Comment (grsterin) [reply src/ci/bin/test.inc.sh:818]: sudo worked, but default pip on Ubuntu 20.04 was referring to python2 version. Once we drop Travis `pip3 install "${pip_package}" "$@"` will be the only version.
Also turned out, sudo was not required here.

Comment (kshakir) [reply src/ci/bin/test.inc.sh:818]: > will be the only version

...until we move with the rest of DSP to GitHub actions 😉

**Comment by breilly2** [comment]
I noticed we don't get test details in CircleCI. We might be able to if we configure `store_test_results` with `centaur/target/test-reports`: https://circleci.com/docs/2.0/configuration-reference/#store_test_results

File: src/ci/bin/test.inc.sh
Comment (breilly2) [inline src/ci/bin/test.inc.sh:1027]: I'm not familiar with how AppRole stuff works, but I'm very surprised to see `vault write` in the test script. I guess that's what this says to do, though: https://www.vaultproject.io/docs/auth/approle. Just pointing it out in case it strikes anyone else as strange.

**Comment by breilly2** [comment]
A couple of other observations:
1. Re-running failures seems to be better with CircleCI. If you click through to the details, you can invoke "Rerun Workflow from Failed" to skip re-running tests that succeeded.
2. We currently have the Travis PR build marked as Required in GitHub. Circle has each test suite split out into its own check. This is great if we want finer control over which tests must pass before merging. This is not so great if we want all PR test suites to succeed because we have to do extra maintenance in GitHub as we add/remove test suites.

File: src/ci/bin/test.inc.sh
Comment (grsterin) [reply src/ci/bin/test.inc.sh:1027]: Also here: https://broadworkbench.atlassian.net/browse/DDO-1050?focusedCommentId=43599

**Comment by grsterin** [comment]
> I noticed we don't get test details in CircleCI. We might be able to if we configure `store_test_results` with `centaur/target/test-reports`: https://circleci.com/docs/2.0/configuration-reference/#store_test_results

I think we also don't have this for Travis. This would be nice to have as a separate ticket

**Comment by grsterin** [comment]
> A couple of other observations:
> 
> 1. Re-running failures seems to be better with CircleCI. If you click through to the details, you can invoke "Rerun Workflow from Failed" to skip re-running tests that succeeded.

I like that a bit less, since in Travis, if one of the subbuilds failed after 1 second, you can independently restart it immediately. But in CircleCI you would need to wait for all subbuilds to finish in order for it to let you restart the failed onces.

**Comment by grsterin** [comment]
Tests succeeded, but Github failed to acknowledge that: https://app.circleci.com/pipelines/github/broadinstitute/cromwell/409/workflows/3271b73b-439b-4666-b7dd-08e8fb8ae99c

Most likely because I enabled full CircleCI integration with Github after creating this PR. Will need to check if it works fine for subsequent PRs

**Review by breilly2 — APPROVED** [review]
👍  Looks good. I do want us to try to get the test results parsed by CircleCI to make the results easier to digest. And hopefully we can get them to be reliable enough that we can stop using Travis. 😄

**Pull request merged**
